crates/sui-swarm-config/src/node_config_builder.rs

```diff
@@ -272,6 +272,7 @@ pub struct FullnodeConfigBuilder {
     run_with_range: Option<RunWithRange>,
     policy_config: Option<PolicyConfig>,
     fw_config: Option<RemoteFirewallConfig>,
+    data_ingestion_dir: Option<PathBuf>,
```

Comment:
Are these changes also related to validator integration with `sui-start`?

Reply:
This is needed for the following reason. When starting an indexer, which is needed for GraphQL for example, we need to set up the `data_ingestion_dir` such that the indexer picks up data that the fullnode is dumping in this temp folder. Without these changes, the indexer would not start to ingest the data.

`sui-test-validator` did this in a bit of a different way due to using `TestClusterBuilder` to instantiate a network.